packages/engine/paima-sm/src/index.ts

```diff
@@ -358,25 +387,64 @@ async function processScheduledData(
   latestChainData: ChainData,
   DBConn: PoolClient,
   gameStateTransition: GameStateTransitionFunction,
-  randomnessGenerator: Prando
+  randomnessGenerator: Prando,
+  precompiles: Precompiles['precompiles'],
+  indexForEvent: (txHash: string) => number
 ): Promise<number> {
   const scheduledData = await getScheduledDataByBlockHeight.run(
     { block_height: latestChainData.blockNumber },
     DBConn
   );
+
+  // just in case there are two timers in the same block with the same exact contents.
+  let timerIndexRelativeToBlock = -1;
+
+  const networks = await GlobalConfig.getInstance();
+
   for (const data of scheduledData) {
     try {
+      let txHash;
+
+      const userAddress = data.precompile ? precompiles[data.precompile] : SCHEDULED_DATA_ADDRESS;
+
+      if (data.precompile && !precompiles[data.precompile]) {
+        doLog(`[paima-sm] Precompile for scheduled event not found ${data.precompile}. Skipping.`);
+        continue;
+      }
+
+      let caip2 = '';
```

Comment:
Is it okay that we leave this as empty string in one of the code paths? (being unsure about this is why I prefer IIFE over this kind of `let` approach)

Reply:
Well, it's empty if it's a timer... Since there is no specific network in that case. I guess we could otherwise make the property optional and just not include it in that case, but I'm not sure it makes much of a difference.

Reply:
I re-wrote it as an IIFE anyway though.